### PR TITLE
Adds documentation for bundling catalog resources in a zip file

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: guide/ops/catalog/index.md [hunk 1]
```diff
@@ -371,6 +372,121 @@ increment an internal version number for the catalog item.
 When referencing a blueprint, if a version number is not specified 
 the latest non-snapshot version will be loaded when an entity is instantiated.
 
+### Bundling Catalog Resources
+
+When deploying a blueprint, it is possible to deploy as an OSGI bundle with additional resources scoped to
```
Comment (drigodwin) [inline guide/ops/catalog/index.md]: possible to deploy it as

File: guide/ops/catalog/index.md [hunk 2]
```diff
@@ -371,6 +372,121 @@ increment an internal version number for the catalog item.
...
+this bundle. This is particularly useful when, for example, deploying a Tomcat server with a `.war` file
+which is deployed using a classpath path such as `classpath://mywar.war`.
+
+In this example, we will create a simple `server` catalog item, bundled with a simple text file
```
Comment (drigodwin) [inline guide/ops/catalog/index.md]: Missing full stop

File: guide/ops/catalog/index.md [hunk 3]
```diff
@@ -371,6 +372,121 @@ increment an internal version number for the catalog item.
...
+~~~ yaml
+location: localhost
+services:
+- type: script-server
+~~~
+
+And we can now inspect the contents of the file copied to the server
+
+~~~ bash
+$ cat /tmp/brooklyn-martin/apps/nl9djqbq2i/entities/EmptySoftwareProcess_g52gahfxnt/files/myfile.txt
```
Comment (drigodwin) [inline guide/ops/catalog/index.md]: The preference is now to not include a `$` at the start of the line but, if required, to split the command and response into two blocks

File: guide/ops/catalog/index.md [hunk 4]
```diff
@@ -371,6 +372,141 @@ increment an internal version number for the catalog item.
...
+  - id: script-server
+    itemType: entity
+    item:
+      type: server
+      brooklyn.config:
+        files.runtime:
+          classpath://myfile.txt: files/myfile.txt
+~~~
+
+The `bundle: script-server` line specifies the OSGI buldle name for this blueprint. Any resources included
```
Comment (tbouron) [inline guide/ops/catalog/index.md]: s/buldle/bundle/

**Comment by nakomis** [comment]
DO NOT MERGE. zip uploading is currently WIP feature

**Comment by sjcorbett** [comment]
@nakomis can you rebase this on master?

File: guide/ops/catalog/index.md [hunk 5]
```diff
@@ -371,6 +372,155 @@ increment an internal version number for the catalog item.
...
+in this bundle will be accessible on the classpath, but will be scoped to this bundle. This prevents an
+issue where multiple bundles include the same resource.
+
+To create the bundle, simply zip the `.bom` and `.txt` files as follows:
+
+~~~ bash
+zip script-server.zip catalog.bom myfile.txt
+~~~
+
+Currently the only supported method for uploading the bundle to the server is via the API, e.g. using
```
Comment (sjcorbett) [inline guide/ops/catalog/index.md]: This isn't true any more. You can `br catalog add script-server.zip` too.

Comment (tbouron) [inline guide/ops/catalog/index.md]: Actually, you don't even have to zip the files, you can pass directly the folder to the `br` command

Comment (sjcorbett) [inline guide/ops/catalog/index.md]: @tbouron cool. Can you give an example? I get an error when I try it:
```
$ br catalog add .
Post http://localhost:8081/v1/catalog: read .: is a directory
$ cd .. && br catalog add bptest
Post http://localhost:8081/v1/catalog: read bptest: is a directory
```

Comment (sjcorbett) [inline guide/ops/catalog/index.md]: My mistake, adding a directory works great.

**Comment by nakomis** [comment]
Closing in favour of #180

**Pull request closed without merging**